sdk/src/main/java/com/hedera/hashgraph/sdk/Client.java

```diff
@@ -1004,7 +1005,15 @@ public synchronized void close() throws TimeoutException {
         try {
             network.close();
             mirrorNetwork.close();
+            executor.shutdown();
+            if (!executor.awaitTermination(30, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+                if (!executor.awaitTermination(60, TimeUnit.SECONDS)) {
```

Comment:
Waiting up until 90 seconds by default is a bit long. After combining the two methods should set a reasonable overall timeout like 30s and decrement time spent between the two `awaitTermination()` methods.